File: z-clients/python/diom/apis/msgs_svix_poller.py
# This file is @generated

from ..internal.api_common import ApiBase
from ..models import (
    ListResponseSvixPollerOut,
    SvixPollerCreateIn,
    SvixPollerCreateOut,
    SvixPollerDeleteIn,
    SvixPollerDeleteOut,
    SvixPollerListIn,
)

from ..models.svix_poller_list_in import _SvixPollerListIn


class MsgsSvixPollerAsync(ApiBase):
    async def create(
        self,
        svix_poller_create_in: SvixPollerCreateIn,
    ) -> SvixPollerCreateOut:
        """Create a Svix poller configuration for a topic."""
        body = svix_poller_create_in.model_dump(exclude_none=True)

        return await self._request_asyncio(
            method="post",
            path="/api/v1.msgs.svix-poller.create",
            body=body,
            response_type=SvixPollerCreateOut,
        )

    async def delete(
        self,
        svix_poller_delete_in: SvixPollerDeleteIn,
    ) -> SvixPollerDeleteOut:
        """Delete a Svix poller configuration."""
        body = svix_poller_delete_in.model_dump(exclude_none=True)

        return await self._request_asyncio(
            method="post",
            path="/api/v1.msgs.svix-poller.delete",
            body=body,
            response_type=SvixPollerDeleteOut,
        )

    async def list(
        self,
        topic: str,
        svix_poller_list_in: SvixPollerListIn = SvixPollerListIn(),
    ) -> ListResponseSvixPollerOut:
        """List Svix poller configurations for a topic."""
        body = _SvixPollerListIn(
            namespace=svix_poller_list_in.namespace,
            topic=topic,
            limit=svix_poller_list_in.limit,
            iterator=svix_poller_list_in.iterator,
        ).model_dump(exclude_none=True)

        return await self._request_asyncio(
            method="post",
            path="/api/v1.msgs.svix-poller.list",
            body=body,
            response_type=ListResponseSvixPollerOut,
        )


class MsgsSvixPoller(ApiBase):
    def create(
        self,
        svix_poller_create_in: SvixPollerCreateIn,
    ) -> SvixPollerCreateOut:
        """Create a Svix poller configuration for a topic."""
        body = svix_poller_create_in.model_dump(exclude_none=True)

        return self._request_sync(
            method="post",
            path="/api/v1.msgs.svix-poller.create",
            body=body,
            response_type=SvixPollerCreateOut,
        )

    def delete(
        self,
        svix_poller_delete_in: SvixPollerDeleteIn,
    ) -> SvixPollerDeleteOut:
        """Delete a Svix poller configuration."""
        body = svix_poller_delete_in.model_dump(exclude_none=True)

        return self._request_sync(
            method="post",
            path="/api/v1.msgs.svix-poller.delete",
            body=body,
            response_type=SvixPollerDeleteOut,
        )

    def list(
        self,
        topic: str,
        svix_poller_list_in: SvixPollerListIn = SvixPollerListIn(),
    ) -> ListResponseSvixPollerOut:
        """List Svix poller configurations for a topic."""
        body = _SvixPollerListIn(
            namespace=svix_poller_list_in.namespace,
            topic=topic,
            limit=svix_poller_list_in.limit,
            iterator=svix_poller_list_in.iterator,
        ).model_dump(exclude_none=True)

        return self._request_sync(
            method="post",
            path="/api/v1.msgs.svix-poller.list",
            body=body,
            response_type=ListResponseSvixPollerOut,
        )

File: z-clients/rust/src/api/msgs_svix_poller.rs
// this file is @generated
use crate::{Configuration, error::Result, models::*};

pub struct MsgsSvixPoller<'a> {
    cfg: &'a Configuration,
}

impl<'a> MsgsSvixPoller<'a> {
    pub(super) fn new(cfg: &'a Configuration) -> Self {
        Self { cfg }
    }

    /// Create a Svix poller configuration for a topic.
    pub async fn create(
        &self,
        svix_poller_create_in: SvixPollerCreateIn,
    ) -> Result<SvixPollerCreateOut> {
        crate::request::Request::new(http::Method::POST, "/api/v1.msgs.svix-poller.create")
            .with_body(svix_poller_create_in)
            .execute(self.cfg)
            .await
    }

    /// Delete a Svix poller configuration.
    pub async fn delete(
        &self,
        svix_poller_delete_in: SvixPollerDeleteIn,
    ) -> Result<SvixPollerDeleteOut> {
        crate::request::Request::new(http::Method::POST, "/api/v1.msgs.svix-poller.delete")
            .with_body(svix_poller_delete_in)
            .execute(self.cfg)
            .await
    }

    /// List Svix poller configurations for a topic.
    pub async fn list(
        &self,
        topic: String,
        svix_poller_list_in: SvixPollerListIn,
    ) -> Result<ListResponseSvixPollerOut> {
        let svix_poller_list_in = SvixPollerListIn_ {
            namespace: svix_poller_list_in.namespace,
            topic,
            limit: svix_poller_list_in.limit,
            iterator: svix_poller_list_in.iterator,
        };

        crate::request::Request::new(http::Method::POST, "/api/v1.msgs.svix-poller.list")
            .with_body(svix_poller_list_in)
            .execute(self.cfg)
            .await
    }
}
